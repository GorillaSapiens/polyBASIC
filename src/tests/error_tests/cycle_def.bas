def a[x] b[x]
def b[x] e[x] + f[x] + c[x]
def c[x] g[x] + d[x]
def d[x] c[x] + h[x]
def e[x] a[x] + f[x]
def f[x] g[x]
def g[x] f[x]
def h[x] g[x] + d[x]
