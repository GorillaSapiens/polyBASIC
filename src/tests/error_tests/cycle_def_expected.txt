ERROR: DEFINITION CYCLE DISCOVERED
   ❮e❯ @5:22 -> ❮a❯ @1:1 -> ❮b❯ @2:15 -> ❮e❯ @5:22
   ❮d❯ @4:22 -> ❮c❯ @3:29 -> ❮d❯ @4:22
   ❮d❯ @4:22 -> ❮h❯ @8:15 -> ❮d❯ @4:22
   ❮g❯ @7:15 -> ❮f❯ @6:22 -> ❮g❯ @7:15
